### PR TITLE
feat: realize basic routing

### PR DESCRIPTION
**Pull request by Finnchick**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -3,15 +3,30 @@ import Header from './commonComponents/Header/Header';
 import AppWrapper from './commonComponents/AppWrapper/AppWrapper';
 import Search from './commonComponents/Search/Search';
 import './sass/App.sass';
+import {BrowserRouter, Route, Switch} from "react-router-dom";
+import BeerPage from "./commonComponents/BeerPage/BeerPage";
+import FavoritesPage from "./commonComponents/FavoritesPage/FavoritesPage";
```
Comment (Dekim12) [inline src/App.jsx:8]: it's better to separate pages from components. Also, only common components for all pages should be in the commonComonents folder.  Components that are related only to one page should be in the folder components for the target page

File: src/App.jsx [hunk 2]
```diff
@@ -3,15 +3,30 @@ import Header from './commonComponents/Header/Header';
...
+        <Switch>
+            <Route path="/beer">
+                <BeerPage beerName={"Mocha"}/>
+            </Route>
+            <Route path='/favorites'>
+                <FavoritesPage/>
+            </Route>
+            <Route path="/">
+                <Search />
+                <AppWrapper />
```
Comment (Dekim12) [inline src/App.jsx:24]: AppWrapper is not appropriate here. Here should be only HomePage with search inside it

File: src/commonComponents/AppWrapper/AppWrapper.jsx [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import React, { useEffect, useState } from 'react';
 import BeerCard from '../BeerCard/BeerCard';
-import './AppWrapper.sass';
+import './style.sass';
 import Loading from '../Loading/Loading';
+
 /* eslint-disable react/prop-types */
```
Comment (Dekim12) [inline src/commonComponents/AppWrapper/AppWrapper.jsx:6]: add rule to eslint file

File: src/commonComponents/AppWrapper/AppWrapper.jsx [hunk 4]
```diff
@@ -11,36 +12,40 @@ function AppWrapper() {
   //       setData(data);
```
Comment (Dekim12) [inline src/commonComponents/AppWrapper/AppWrapper.jsx:12]: please, avoid using commented parts of code in the PRs

File: src/commonComponents/AppWrapper/AppWrapper.jsx [hunk 5]
```diff
@@ -11,36 +12,40 @@ function AppWrapper() {
...
-    setLoading(false);
+   // setLoading(false);
   }, []);
 
   return (
-    <div className="AppWrapper">
-      {loading ? (
-        <Loading />
+    <div className="appWrapper">
+
```
Comment (Dekim12) [inline src/commonComponents/AppWrapper/AppWrapper.jsx:27]: remove empty line

File: src/commonComponents/AppWrapper/AppWrapper.jsx [hunk 6]
```diff
@@ -11,36 +12,40 @@ function AppWrapper() {
...
+      {dataT ? (
+          <React.Fragment>
+            {dataT.map((item) => {
+              return (
+                  <BeerCard
+                      key={item.id}
+                      id={item.id}
+                      cardHeader={item.name}
```
Comment (Dekim12) [inline src/commonComponents/AppWrapper/AppWrapper.jsx:35]: use destructuring of parameters

File: src/commonComponents/AppWrapper/AppWrapper.jsx [hunk 7]
```diff
@@ -11,36 +12,40 @@ function AppWrapper() {
...
 
   return (
-    <div className="AppWrapper">
-      {loading ? (
-        <Loading />
+    <div className="appWrapper">
+
+      {dataT ? (
+          <React.Fragment>
+            {dataT.map((item) => {
```
Comment (Dekim12) [inline src/commonComponents/AppWrapper/AppWrapper.jsx:30]: define it before return statement

File: src/commonComponents/AppWrapper/AppWrapper.jsx [hunk 8]
```diff
@@ -11,36 +12,40 @@ function AppWrapper() {
...
-                information={item.tagline}
-                imageUrl={item.image_url}
-              />
-            );
-          })}
-        </React.Fragment>
+          <Loading />
+
       )}
+
```
Comment (Dekim12) [inline src/commonComponents/AppWrapper/AppWrapper.jsx:46]: remove empty line

**Comment by Finnchick** [comment]
Done

File: .gitignore [hunk 9]
```diff
@@ -21,3 +21,4 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply .gitignore:23]: Done

File: src/commonComponents/Loading/Loading.jsx [hunk 10]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
-import './Loading.css';
+import './style.sass';
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/Loading/Loading.jsx:2]: Done

File: src/commonComponents/BeerPage/BeerPage.jsx [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+                    <h1>{beerName}</h1>
+                    <h4>Post Modern Classic. Spiky. Tropical. Hoppy.</h4>
+                    <BeerCardButton buttonText="Add to favorites" buttonClass={'beerCardButton'}/>
+                    <p>Our flagship beer that kick started the craft beer revolution. This is James and Martins
+                        original take on an American IPA, subverted with punchy New Zealand hops. Layered with new world
+                        hops to create an all-out riot of grapefruit, pineapple and lychee before a spiky,
+                        mouth-puckering bitter finish.</p>
+                </div>
+                <div className="beer-avatar-box">
+                    <img src="https://images.punkapi.com/v2/192.png" alt="Beer Image" className="beerAvatar"/>
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerPage/BeerPage.jsx]: It's only placeholder before redux implementation. But I understood, will do it for other URLs

File: src/commonComponents/BeerCardButton/BeerCardButton.jsx [hunk 12]
```diff
@@ -1,8 +1,21 @@
 import React from 'react';
-import './BeerCardButton.sass';
+import './style.sass';
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCardButton/BeerCardButton.jsx]: Done

File: src/commonComponents/BeerCard/BeerCard.jsx [hunk 13]
```diff
@@ -1,22 +1,28 @@
...
+function BeerCard({ id, cardHeader, information, imageUrl }) {
+
+
   return (
-    <div className="BeerCard">
+    <div className="beerCard">
       <img className="avatar" src={imageUrl} alt="Beer image"></img>
       <h2>{cardHeader}</h2>
       <p>{information}</p>
       <div className="beercardButtons">
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCard/BeerCard.jsx]: Done

File: src/commonComponents/Search/Search.jsx [hunk 14]
```diff
@@ -3,30 +3,26 @@ import FilterResults from '../FilterResults/FilterResults';
...
+  const [resultsActive, setResultsActive] = useState(false);
 
   return (
     <div>
       <div className="search-box">
-        <script
-          src="https://kit.fontawesome.com/a8db9ba565.js"
-          crossOrigin="anonymous"
-        ></script>
         <button className="btn-search" onClick={() => setResultsActive(true)}>
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/Search/Search.jsx]: changed all css styles to Hyphen Delimited Strings

File: src/commonComponents/Search/Search.jsx [hunk 15]
```diff
@@ -3,30 +3,26 @@ import FilterResults from '../FilterResults/FilterResults';
 //import loading from './icons8-search.svg';
 import { faSearch } from '@fortawesome/free-solid-svg-icons';
 
-import './Search.sass';
+import './style.sass';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 function Search() {
-  const [resultsAcitve, setResultsActive] = useState(false);
+  const [resultsActive, setResultsActive] = useState(false);
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/Search/Search.jsx]: changed to areResultsActive

File: src/commonComponents/Menu/Menu.jsx [hunk 16]
```diff
@@ -1,24 +1,30 @@
...
-  );
+function Menu({active, setActive}) {
+    return (
+        <nav
+            className={active ? 'menuActive' : 'menu'}
+            onClick={() => setActive(false)}
+        >
+            <ul className="menu__list">
+                <Link to="/">
+                    <li className="menu__item">
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/Menu/Menu.jsx]: changed all css class names to Hyphen Delimited Strings

File: src/commonComponents/Header/Header.jsx [hunk 17]
```diff
@@ -1,19 +1,19 @@
...
-import './Header.sass';
+import './style.sass';
 //import BurgerButton from "./commonComponents/BurgerButton.jsx"
 import BurgerButton from '../BurgerButton/BurgerButton';
 import Menu from '../Menu/Menu';
 
 function Header() {
-  const [buttonActive, setButtonActive] = useState(false);
+  const [isMenuShown, ShowMenu] = useState(false);
   return (
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/Header/Header.jsx:8]: Done

File: src/commonComponents/Header/Header.jsx [hunk 18]
```diff
@@ -1,19 +1,19 @@
...
     <div className="Header">
-      <BurgerButton active={buttonActive} setActive={setButtonActive} />
-      <Menu active={buttonActive} setActive={setButtonActive}></Menu>
+      <BurgerButton active={isMenuShown} setActive={ShowMenu} />
+      <Menu active={isMenuShown} setActive={ShowMenu}/>
       <h1 className="Logo"> Beer Catalagoue </h1>
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/Header/Header.jsx]: changed all css class names to Hyphen Delimited Strings

File: src/commonComponents/Header/Header.jsx [hunk 19]
```diff
@@ -1,19 +1,19 @@
 import React, { useState } from 'react';
-import './Header.sass';
+import './style.sass';
 //import BurgerButton from "./commonComponents/BurgerButton.jsx"
 import BurgerButton from '../BurgerButton/BurgerButton';
 import Menu from '../Menu/Menu';
 
 function Header() {
-  const [buttonActive, setButtonActive] = useState(false);
+  const [isMenuShown, ShowMenu] = useState(false);
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/Header/Header.jsx:7]: Done

File: src/commonComponents/FavoritesPage/FavoritesPage.jsx [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+
+/* eslint-disable react/prop-types */
+function FavoritesPage() {
+
+    return (
+        <div></div>
+    )
+        ;
+}
+
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/FavoritesPage/FavoritesPage.jsx]: Done

File: src/commonComponents/BeerPage/BeerPage.jsx [hunk 21]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import BeerCardButton from "../BeerCardButton/BeerCardButton";
+import './style.sass'
+
+/* eslint-disable react/prop-types */
+function BeerPage({beerName}) {
+
+    return (
+        <div className="beerPageWrapper">
+            <div className="beer-card">
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerPage/BeerPage.jsx]: changed all css class names to Hyphen Delimited Strings

File: src/commonComponents/BeerPage/BeerPage.jsx [hunk 22]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import BeerCardButton from "../BeerCardButton/BeerCardButton";
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerPage/BeerPage.jsx]: Done

File: src/commonComponents/BeerCard/style.sass [hunk 23]
```diff
@@ -12,14 +12,14 @@
     height: 300px
     margin: 50px 50px
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCard/style.sass]: Done

File: src/commonComponents/BeerPage/BeerPage.jsx [hunk 24]
```diff
@@ -0,0 +1,69 @@
...
+/* eslint-disable react/prop-types */
+function BeerPage({beerName}) {
+
+    return (
+        <div className="beerPageWrapper">
+            <div className="beer-card">
+                <div className="beerMinimumInfo">
+                    <h1>{beerName}</h1>
+                    <h4>Post Modern Classic. Spiky. Tropical. Hoppy.</h4>
+                    <BeerCardButton buttonText="Add to favorites" buttonClass={'beerCardButton'}/>
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerPage/BeerPage.jsx]: changed from {} to ''

File: src/commonComponents/BeerCard/BeerCard.jsx [hunk 25]
```diff
@@ -1,22 +1,28 @@
...
         <BeerCardButton buttonText={'Open'} buttonClass={'beerCardButton'} />
         <BeerCardButton
-          buttonText={'Favourite'}
+            id = {id}
+          buttonText='Favourite'
           buttonClass={'beerCardButton'}
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCard/BeerCard.jsx:17]: changed from {} to ''

File: src/commonComponents/BeerCard/BeerCard.jsx [hunk 26]
```diff
@@ -1,22 +1,28 @@
...
+
+
   return (
-    <div className="BeerCard">
+    <div className="beerCard">
       <img className="avatar" src={imageUrl} alt="Beer image"></img>
       <h2>{cardHeader}</h2>
       <p>{information}</p>
       <div className="beercardButtons">
         <BeerCardButton buttonText={'Open'} buttonClass={'beerCardButton'} />
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCard/BeerCard.jsx:13]: changed from {} to ''

File: src/commonComponents/AppWrapper/AppWrapper.jsx (hunk 6 above)
Comment (Finnchick) [reply src/commonComponents/AppWrapper/AppWrapper.jsx]: Done

File: src/commonComponents/AppWrapper/AppWrapper.jsx (hunk 4 above)
Comment (Finnchick) [reply src/commonComponents/AppWrapper/AppWrapper.jsx]: Cleared all comments

File: src/App.jsx (hunk 1 above)
Comment (Finnchick) [reply src/App.jsx]: Done

File: src/commonComponents/BurgerButton/BurgerButton.jsx [hunk 27]
```diff
@@ -1,11 +1,11 @@
 /* eslint-disable react/prop-types */
 import React from 'react';
-import './BurgerButton.sass';
+import './style.sass';
 
 function BurgerButton({ active, setActive }) {
   return (
     <div
-      className={active ? 'BurgerButton change' : 'BurgerButton'}
+      className={active ? 'burgerButton change' : 'burgerButton'}
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BurgerButton/BurgerButton.jsx:7]: used in burgerButton and Menu component

File: src/commonComponents/Menu/Menu.jsx [hunk 28]
```diff
@@ -1,24 +1,30 @@
...
-          <FontAwesomeIcon icon={faStar} /> Favorietes
-        </li>
-      </ul>
-    </nav>
-  );
+function Menu({active, setActive}) {
+    return (
+        <nav
+            className={active ? 'menuActive' : 'menu'}
+            onClick={() => setActive(false)}
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/Menu/Menu.jsx]: Done

File: src/commonComponents/BeerCardButton/BeerCardButton.jsx [hunk 29]
```diff
@@ -1,8 +1,21 @@
...
+import {useHistory} from "react-router-dom";
+
+
 /* eslint-disable react/prop-types */
-function BeerCardButton({ buttonText, buttonClass }) {
-  return <button className={buttonClass}>{buttonText}</button>;
+function BeerCardButton({ id, buttonText, buttonClass}) {
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCardButton/BeerCardButton.jsx]: Done

File: src/App.jsx (hunk 2 above)
Comment (Finnchick) [reply src/App.jsx]: Done

File: src/commonComponents/BeerCardButton/BeerCardButton.jsx [hunk 30]
```diff
@@ -1,8 +1,21 @@
...
+  let history = useHistory()
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCardButton/BeerCardButton.jsx]: changed structure. Now I'm creating history object in main page component and pass callback in map function. Check this realization please)

File: src/commonComponents/BeerCard/BeerCard.jsx [hunk 31]
```diff
@@ -1,22 +1,28 @@
...
           buttonClass={'beerCardButton'}
+
         />
       </div>
     </div>
   );
 }
 
-export default BeerCard;
+export default BeerCard;
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCard/BeerCard.jsx:24]: Checked

File: src/commonComponents/AppWrapper/AppWrapper.jsx (hunk 8 above)
Comment (Finnchick) [reply src/commonComponents/AppWrapper/AppWrapper.jsx]: Done

File: src/commonComponents/AppWrapper/AppWrapper.jsx (hunk 7 above)
Comment (Finnchick) [reply src/commonComponents/AppWrapper/AppWrapper.jsx]: Done

File: src/commonComponents/AppWrapper/AppWrapper.jsx (hunk 5 above)
Comment (Finnchick) [reply src/commonComponents/AppWrapper/AppWrapper.jsx]: Done

File: src/commonComponents/AppWrapper/AppWrapper.jsx (hunk 3 above)
Comment (Finnchick) [reply src/commonComponents/AppWrapper/AppWrapper.jsx]: Done

File: src/commonComponents/BeerCard/style.sass [hunk 32]
```diff
@@ -1,6 +1,6 @@
-.BeerCard 
+.beerCard
     box-sizing: border-box
-    display: flex;
+    display: flex
     flex-direction: column
     color: black
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCard/style.sass]: Done

File: src/commonComponents/BeerCard/BeerCard.jsx [hunk 33]
```diff
@@ -1,22 +1,28 @@
...
+
 /* eslint-disable react/prop-types */
-function BeerCard({ cardHeader, information, imageUrl }) {
+function BeerCard({ id, cardHeader, information, imageUrl }) {
+
+
   return (
-    <div className="BeerCard">
+    <div className="beerCard">
       <img className="avatar" src={imageUrl} alt="Beer image"></img>
```
[the remark this answers is not in the document]
Comment (Finnchick) [reply src/commonComponents/BeerCard/BeerCard.jsx]: Done

**Review by Dekim12 — APPROVED** [review]
Good job!

**Pull request merged**